AmplifyPlugins/Core/AWSPluginsCoreTests/ServiceConfiguration/AmplifyAWSServiceConfigurationTests.swift

```diff
@@ -26,4 +38,30 @@ class AmplifyAWSServiceConfigurationTests: XCTestCase {
         XCTAssertEqual(expectedSystem, userAgentParts[1])
         XCTAssertEqual(expectedLocale, userAgentParts[2])
     }
+
+    /// Test adding a new platform to AmplifyAWSServiceConfiguration
+    ///
+    /// - Given: Amplify library
+    /// - When:
+    ///    - I add a new platform to the AmplifyAWSServiceConfiguration
+    /// - Then:
+    ///    - AmplifyAWSServiceConfiguration should be configured properly with the new platform added.
+    ///
+    func testAddNewPlatform() {
+        AmplifyAWSServiceConfiguration.addUserAgentPlatform(.flutter, version: "1.1")
+        let currentSystemName = UIDevice.current.systemName.replacingOccurrences(of: " ", with: "-")
+        let currentSystemVersion = UIDevice.current.systemVersion
+        let expectedLocale = Locale.current.identifier
+        let expectedSystem = "\(currentSystemName)/\(currentSystemVersion)"
+
+        let configuration = AmplifyAWSServiceConfiguration()
+
+        XCTAssertNotNil(configuration.userAgent)
+        let userAgentParts = configuration.userAgent.components(separatedBy: " ")
+        XCTAssertEqual(4, userAgentParts.count)
+        XCTAssert(userAgentParts[0].starts(with: AmplifyAWSServiceConfiguration.Platform.flutter.rawValue))
```

Comment:
You should be able to assert the entire platform & version value right?

Reply:
Changed